functest/extract_test.go

```diff
@@ -0,0 +1,208 @@
+		// 5. For all cases, make sure we have a Dockerfile extracted
+		// at the vortex of the extraction.
+
+		oldDir, err := os.Getwd()
+		if err != nil {
+			t.Fatal("Error getting current directory: ", err)
+		}
+
+		err = os.Chdir(projectDir)
+		if err != nil {
+			t.Fatal("Error changing directory: ", err)
+		}
+		mounts, _ := cmd.GetEnvVar("APPSODY_MOUNTS")
+		pDir, _ := cmd.GetEnvVar("APPSODY_PROJECT_DIR")
+
+		err = os.Chdir(oldDir)
+		if err != nil {
+			t.Fatal("Error changing directory: ", err)
+		}
+
+		log.Println("Stack mounts:", mounts)
+		if pDir == "" {
+			pDir = "/project"
+		}
+		log.Println("Stack's project dir:", pDir)
+
+		mountlist := strings.Split(mounts, ";")
+		for _, mount := range mountlist {
+			log.Println("mount:", mount)
+			src := strings.Split(mount, ":")[0]
+			dest := strings.Split(mount, ":")[1]
+			if !strings.HasPrefix(dest, pDir) {
+				log.Println("Skipping un-extracted content", src, "and", dest)
+				continue
+			}
+			remote := strings.Replace(dest, pDir, extractDir, -1)
+			var local string
+			homeDir, homeErr := os.UserHomeDir()
+			if homeErr != nil {
+				t.Fatal("Unable to find user home location:", homeErr)
+			}
+			if strings.HasPrefix(src, "~") {
+				local = strings.Replace(src, "~", homeDir, 1)
+			} else {
+				local = projectDir + "/" + src
+			}
+			log.Println("local: ", local)
+			log.Println("remote: ", remote)
+
+			fileInfoLocal, err := os.Lstat(local)
+			if err != nil {
+				t.Fatal("Mount inspection error:", err)
+			}
+			if fileInfoLocal.IsDir() {
+				localData, err := ioutil.ReadDir(local)
+				if err != nil {
+					t.Fatal("Mount inspection error:", err)
+				}
+				extractData, err := ioutil.ReadDir(remote)
+				if err != nil {
+					t.Fatal("Mount inspection error:", err)
+				}
+				localContent := []string{}
+				extractContent := []string{}
+				for _, file := range localData {
+					localContent = append(localContent, file.Name())
+				}
+				for _, file := range extractData {
+					extractContent = append(extractContent, file.Name())
+				}
+				if !reflect.DeepEqual(extractContent, localContent) {
+					t.Fatal("Extraction failure, ", local, " is not extracted into ", remote)
+				} else {
+					log.Println("Folder contents match.")
+				}
+			} else {
+				fileInfoRemote, err := os.Lstat(remote)
+				if err != nil {
+					t.Fatal("Mount inspection error:", err)
+				}
+				dSize := fileInfoRemote.Size()
+				lSize := fileInfoLocal.Size()
+				if lSize != dSize {
+					t.Fatal("Extraction failure, ", local, " is not extracted into ", remote, " properly: source file size: ", lSize, " destination file size: ", dSize)
+				} else {
+					log.Println("File sizes match.")
+				}
+
+			}
+
+		}
+		dockerFile := filepath.Join(extractDir, "Dockerfile")
+		_, err = exists(dockerFile)
+		if err != nil {
+			t.Fatal("Extraction failure, Dockerfile was not extracted into ", extractDir)
+		}
+	}
+}
+
+// private exists fn.
+// TODO: Can this be re-used from cmd/utils.go?
```

Comment:
Yes - we should reuse the one in utils. Please make it public.

Reply:
@chilanti - raised in #328 , PTAL, thanks!